library/std/src/sys/windows/path.rs

```diff
@@ -141,3 +162,55 @@ fn parse_next_component(path: &OsStr, verbatim: bool) -> (&OsStr, &OsStr) {
         None => (path, OsStr::new("")),
     }
 }
+
+/// Gets an absolute path with a verbatim prefix.
+///
+/// This allows bypassing the legacy `MAX_PATH` limits.
+pub(crate) fn to_verbatim(path: &Path) -> io::Result<Vec<u16>> {
+    let mut path = to_u16s(path)?;
+
+    // Early return for paths that are already verbatim.
+    if path.starts_with(VERBATIM_PREFIX) || path.starts_with(NT_PREFIX) {
```

Comment:
Device paths starting with `\\.\` should be treated specially as well here and below

Reply:
But device paths are also limited to `MAX_PATH`, so surely they need to be handled too?

Reply:
With the current code `\\.\COM1` is converted to `\\?\UNC\.\COM1`, similarly `\\.\BootPartition\` to `\\?\UNC\.\BootPartition`. I don't have a Windows machine nearby right now to test. Do paths like this work?

Reply:
I thought I had a test to make sure that worked correctly? E.g.  `check(r"\\.\CON", r"\\?\CON");`

Reply:
Ah, you are right of course, sorry I missed that. Looks good to me.